crates/red_knot_python_semantic/src/types.rs

```diff
@@ -325,6 +325,8 @@ pub enum Type<'db> {
     ModuleLiteral(File),
     /// A specific class object
     ClassLiteral(ClassLiteralType<'db>),
+    // The set of all class objects that are subclasses of the given class (C), spelled `type[C]`.
+    Type(ClassLiteralType<'db>),
```

Comment:
Other sensible, maybe less confusing names might be
* `Type::TypeOf` because `type[…]` resembles "type of …"
* `Type::MetaType`
* …

Reply:
If we don't feel that we need to stay tied to a name that's close to the syntactic form used in annotations, I think a good name for what this type actually means would be `Type::SubclassOf`. I think I would be happy with prioritizing semantic clarity, and relying on the doc comment to clarify that this represents `type[]`.

Reply:
`SubclassOf` is a decent name, I'm happy to go with that. I also like `MetaType`.

Reply:
And it's nice that `issubclass(cls, classinfo)` imposes a `Type::SubclassOf(classinfo)` constraint on `cls`.